traffic_ops/traffic_ops_golang/server/update.go

```diff
@@ -296,9 +296,16 @@ func UpdateHandlerV4(w http.ResponseWriter, r *http.Request) {
 		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
 		return
 	}
-	userErr, sysErr, statusCode := dbhelpers.CheckIfCurrentUserHasCdnLock(inf.Tx.Tx, string(cdnName), inf.User.UserName)
-	if userErr != nil || sysErr != nil {
-		api.HandleErr(w, r, inf.Tx.Tx, statusCode, userErr, sysErr)
+
+	_, hasConfigUpdatedBoolParam := inf.Params["updated"]
+	_, hasRevalUpdatedBoolParam := inf.Params["reval_updated"]
+	_, hasConfigApplyTimeParam := inf.Params["config_apply_time"]
+	_, hasRevalidateApplyTimeParam := inf.Params["revalidate_apply_time"]
+	// Allow `apply_time` changes when the CDN is locked, but not `updated`
+	canIgnoreLock := (hasConfigApplyTimeParam || hasRevalidateApplyTimeParam) && !hasConfigUpdatedBoolParam && !hasRevalUpdatedBoolParam
+	userDoesntHaveLockErr, sysErr, statusCode := dbhelpers.CheckIfCurrentUserHasCdnLock(inf.Tx.Tx, string(cdnName), inf.User.UserName)
```

Comment:
You just need to perform this check if `canIgnoreLock` is `false`. Else, you don't even need to run this query.